react/src/context/PostHogProvider.tsx

```diff
@@ -45,7 +43,7 @@ export function PostHogProvider({ children, client, apiKey, options }: WithOptio
     // This is used to prevent double initialization when running under React.StrictMode
     // We're not storing a simple boolean here because we want to be able to detect if the
     // apiKey or options have changed.
-    const [alreadyInitialized, setAlreadyInitialized] = useState<AlreadyInitialized>(false)
+    const previousInitializationRef = useRef<PreviousInitialization | null>(null)
```

Comment:
Since you're not matching against the `ref` object directly, you're checking the internal props, this makes more sense as a `useState`, don't you think? Does that break SSR?

Reply:
Oh, how I'd love to have a conversation with Guillermo/someone at Meta, this is just nonsense.

React: be reactive
Also React: force you to use declarative refs to work around it being reactive